gradoop-flink/src/main/java/org/gradoop/flink/io/impl/csv/indexed/IndexedCSVDataSource.java

```diff
@@ -97,9 +98,10 @@ public GraphCollection getGraphCollection() throws IOException {
       source.readDistributed(getMetaDataPath(), getConfig());
 
     ExecutionEnvironment env = getConfig().getExecutionEnvironment();
-    EPGMGraphHeadFactory graphHeadFactory = getConfig().getGraphHeadFactory();
-    EPGMVertexFactory vertexFactory = getConfig().getVertexFactory();
-    EPGMEdgeFactory edgeFactory = getConfig().getEdgeFactory();
+    GraphCollectionFactory factory = getConfig().getGraphCollectionFactory();
+    GraphHeadFactory<EPGMGraphHead> graphHeadFactory = factory.getGraphHeadFactory();
```

Comment:
No more need to declare the factories at this point. Call and pass them in the constructor of the fitting map function (equal to CSVDataSource).

 E.g., 
For `new CSVLineToGraphHead(graphHeadFactory)` -> `new CSVLineToGraphHead(factory.getGraphHeadFactory())`

Reply:
I've done this small change.